### ISSUE TITLE
BelongsToMany Fields not Showing on Create/Update Screens

### ISSUE THREAD

**Issue by bahamagician** [laravel/nova-issues#292]
I'm having an issue with not being able to show the belongstomany or hasmany fields on the create/update screens.  

For example I have a product and category relation in which a product can belong to multiple categories and categories can belong to multiple products.  I'm using a category_product pivot table for this.  

The docs lead me to believe this is possible.  Below is a quote from the relationship docs:

> When a BelongsToMany field is shown on a resource creation / update screen, a drop-down selection menu or search menu will display the "title" of the resource. For example, a Role resource may use the name attribute as its title. Then, when the resource is shown in a BelongsToMany selection menu, that attribute will be displayed:

**Below is my code for the product field definitions:**

```
    public function fields(Request $request)
    {
        return [
            ID::make()->sortable(),
            Text::make('Title'),
            HasMany::make('Images'),
            BelongsToMany::make('Categories', 'category')->searchable()
        ];
    }
```

**And here is my code for my category field definitions:**

```
 public function fields(Request $request)
    {
        return [
            ID::make()->sortable(),
            Text::make('Title'),
            BelongsToMany::make('Products')->searchable()
        ];
    }
```
And yet when I visit the edit or create screens for Category or Product, I don't get the fields:

**Update Product Page**
![image](https://user-images.githubusercontent.com/13170078/44738018-9e892c00-aac1-11e8-97db-55a54af388e4.png)

**Create Category Page**
![image](https://user-images.githubusercontent.com/13170078/44738059-beb8eb00-aac1-11e8-8cde-73ae295a0410.png)

**And just in-case, here's my full repo:**
https://github.com/bahamagician/nova-testing

Any help would be greatly appreciated.  Thanks.

**Comment by megan-starr9** [comment]
Fairly certain this is a misleading line in the docs - you can view/modify your relationships on the "View Details" screens now, but as far as I know, not the edit/add screens

**Comment by bahamagician** [comment]
Based on my twitter thread with Taylor it seems that this is possible but I'm doing something wrong:

[https://twitter.com/bahamagician/status/1034222817619791872](https://twitter.com/bahamagician/status/1034222817619791872)

**Comment by megan-starr9** [comment]
oh, wow.  thanks! Guess I'll follow this thread and start seein if I can sort it out too!

**Comment by bradykeeling** [comment]
I'm having the same issue as well... I'll keep an eye on that thread too!

**Comment by pabloguerrez** [comment]
same here

**Comment by megan-starr9** [comment]
I wonder if the screen they are referring to is the actual add/modify relationship screen instead of adding/modifying the parent resource.... because you do see the title dropdown when you go in to actually modify a parent row.  The issue is not seeing the actual relationship table when modifying or editing the parent resource.  if that makes sense.

**Comment by bahamagician** [comment]
Thanks for creating the pull request.  I've also spent days trying to figure it out myself, so any clarity would be greatly appreciated.

**Comment by megan-starr9** [comment]
Documentation update merged!  Looks like these should only show up on view, I think this can be closed

**Comment by bahamagician** [comment]
Thanks for helping :)

**Comment by elrufoelrufo** [comment]
Same here. Did you find any workaround to do it work?

**Comment by niladam** [comment]
@davidhemphill - do you know if there are any plans on supporting adding multiple items in the same screen ?

**Comment by terasupernova** [comment]
@bahamagician are you able to create the custom field? I faced some problem when create and update the form. 
My parent table is Offer and Rule table
Relationship table is Offer-Rule

public function rules()  
{
```
 return $this->belongsToMany(Rule::class, 'offer_rule')->withPivot('value')
                ->using(ValuePivot::class);
```
}

When i create/update the table my custom field cannot pass through the validateForCreation() function in ResourceStoreController or validateForUpdate in ResourceUpdateController. 

<img width="727" alt="screen shot 2019-01-09 at 11 38 12 am" src="https://user-images.githubusercontent.com/8252838/50875394-13228e80-1403-11e9-9f7c-9a9e72124a2d.png">

The image above show the error when i click on create button. 
Do you face the same issue?

Thanks

**Comment by vernard** [comment]
+1

**Comment by denismarinov** [comment]
> @davidhemphill - do you know if there are any plans on supporting adding multiple items in the same screen ?

Would be nice to know if this is going to be a feature some day. 

Also I am curious what workarounds did you do to make it work on the create view

**Comment by dillingham** [comment]
@thanoseleftherakoa I’m working on a package for this if you want to test it out and let me know if it works for your usecase 

https://github.com/dillingham/nova-attach-many

**Comment by dillingham** [comment]
@thanoseleftherakos I’m working on a package for this if you want to test it out and let me know if it works for your usecase 

https://github.com/dillingham/n

**Comment by m-lotze** [comment]
This should be in core. When you edit a resource you should be able to edit all the data and relationships. Not beeing able to edit a belongstomany relationship on an update/create form ist just confusing for the user and adds unnecessary klicks.

**Comment by nlenkowski** [comment]
I agree with @m-lotze. Its awkward that belongsToMany relationships cannot be edited in place when creating and/or editing a resource. It also isn't intuitive that those relationships are managed via the "view" screen instead of the "edit" screen, where one would expect to. 

I just deployed a new site to production recently and the client was both confused and irritated by this. It meant that for each resource (of many) that he had to enter he had to first create the resource, then view it to set the belongsToMany relationships. I have to admit I was even confused the first time I tried to manage the relationship myself. I thought I'd configured the Nova resource wrong initially, but then realized this was just how it worked.

I definitely think this could use some further consideration.

**Comment by matthewjumpsoffbuildings** [comment]
i ran into this recently. it makes absolutely no logical sense from a UI/UX perspective. i thought i had implemented my resource/models incorrectly because i didnt think to click "view" when i wanted to test attaching related items. why on earth would you not be able to do this in the edit/create mode?

**Comment by khaledBawz** [comment]
As developers said, this feature must be considered for the upcoming updates.

**Comment by kamioon** [comment]
Any update or new solution for having BelongsToMany in the Update screen?

**Comment by chendicott** [comment]
Adding a +1 to definitely consider adding this to the update screen as apart of core. Blew a chunk of time wondering why the related items weren't showing up under the edit screen, when logically you're editing the item.

**Comment by iruoy** [comment]
This text made me think this was possible and I just did something wrong. It took me quite a while to find out that it isn't possible because I believe it should definitively be.

![Screen Shot 2020-02-18 at 08 12 36](https://user-images.githubusercontent.com/5859352/74712385-73028000-5226-11ea-8b92-9e97055bea46.png)
https://github.com/laravel/nova-docs/blob/develop/2.0/resources/relationships.md#title-attributes-1

**Comment by packytagliaferro** [comment]
This for sure should be core functionality. If people are paying for this product, it just makes sense to make the UX work as smoothly as possible. I get wanting to always just reply to requests/issues with the whole "we didnt make it work this way", close the issue, never respond again. We probably all do work and want to tell clients that. End of the day though if a client makes a good point and is paying me, I will make them happy. Seems like this is an issue where Nova's clients (us) seem to agree and this should be added. 

I also understand this is not correct forum to make a ton of "Feature Requests" so maybe start something on https://featureupvote.com/ where we can track all the requests and vote them up? Seems like a way to make everyone happy since I love Nova, but finding it hard to see what features are requested or a way to help move them forward.

**Comment by davidhemphill** [comment]
We close feature requests so they don't clog up bugs. We encourage people to upvote the closed issue, since that's one of the signals we use to find stuff to improve on.

**Comment by packytagliaferro** [comment]
@davidhemphill thanks for the response. I will upvote this one then, just wasnt clear how to make a feature request move up. Just a suggestion, but have you all looked at something like https://featureupvote.com/  ? Might be a more clear way (for your team and users)  for people to add feature requests and vote on them? I have used it for Adobe products and its nice way to know that my requests are heard and also not duplicate them, or clog up GitHub since like I said, I figured it was the wrong forum. Just a thought, but thanks for clearing it up.

**Comment by nlenkowski** [comment]
For anyone interested in this feature becoming part of Nova core, please upvote the first post by @bahamagician.

Thanks also for the reply @davidhemphill. It's nice to know that even these closed issues are still being monitored by Nova developers

**Comment by rreynier** [comment]
I spent a whole afternoon fiddling with relationships, models, digging into source code trying to figure out why this relationship was not showing up in the models view.   Intuitively you would never think you need to go to a none-edit view to make a change to a model. It seems like this is likely implemented as is due to technical challenges but from a users perspective we really should consider making this more friendly. Take my upvote :+1:

**Comment by php-writerman** [comment]
https://github.com/dillingham/nova-attach-many

**Comment by core45** [comment]
I totally agree. These relationships should be edited in the create/edit views. It's completely illogical to define them in the 'view' view

**Comment by AforDesign** [comment]
I think this would be one of the biggest improvements.

It's very illogical to edit relations on the 'view' view and hiding them on the update and create views.
I've never seen this structure before in any other system.
Ususally view directs to the front end view if applicable.
There's not a good reason I can think of to have a separate create and update view.

Right now it's cumbersoe but workable for me, but for my non-technical client it's utter madness. 
It makes editing models way too complex.

**Comment by powolnymarcel** [comment]
Did someone found a workaround for this ?

**Comment by StanMenten-bp** [comment]
This is a must-have for an awesome tool as Nova. I'm a little confused why it's not managable via the edit page.

**Comment by agoyli** [comment]
+1

**Comment by iclicksee** [comment]
Why is this issue closed? such bad UX. I wouldn't of used Nova if I new this was a limitation.

**Comment by core45** [comment]
Yes, why? Open it again please

**Comment by AforDesign** [comment]
> Why is this issue closed? such bad UX. I wouldn't of used Nova if I new this was a limitation.

Just one of many limitations and annoyances.
I deeply regret buying and using Nova.

At the time it sounded like an no-brainer:
- created by the creators of Laravel (which truely is briliant) itself
- slick interface
- supporting all relations out of the box
- good documentation (on first sight)

When I bought Nova, I used Backpack, but it didn't support Many-to-Many and MorphToMany relations at that time, which was essential for my project.

In the field, using Nova turned out to be a nightmare to use (for me).
It set me back about a year in developing my project. Hundreds of hours I can't charge my client.
So much struggles, so many conflicts and so little progress on improvement and new features.
Hardly any support. Most issues are ignored or simply closed.
The UI / UX is very impractical, making things hard to grasp. For my client it's totally incomprehensible.
As a result I've been creating custom front end edit options for them (and myself). 


In retrospect I should've stuck with Backpack, but I can't affort to change everything again.
At least now I know why there's no trial or free development version.

Luckely all updates stayed free so far.
It still feels we're on version 0.6 beta, so I guess that's reasonable.

I'm sorry to go off-topic ranting like this, but it's just so frustrating at times.

**Comment by nlenkowski** [comment]
@AforDesign is right, this issue needs further consideration. It really drags the user experience down when creating resources, which is one of the main uses for Nova.

Any word on this @davidhemphill?
